config/config/src/main/java/io/helidon/config/EnumMapperProvider.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.config;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+import io.helidon.config.spi.ConfigMapperProvider;
+
+/**
+ * Built-in mapper for {@code enum}s.
+ *
+ * <p>
+ *     This mapper attempts to match strings in the config source to enum values as follows:
+ *     <ul>
+ *         <li>The mapper treats hyphens ('-') in config strings as underscores when comparing to enum value names.</li>
```

Comment:
Here was my thinking on including hyphens.

In config, hyphens are often used stylistically instead of underscores as for readability _within_ names (`buffer-size`). So it seemed natural to support an equivalence between hyphens for in-name readability style in config sources and underscores as in-name readability style in `enum` value names.

I (at least) do not see dots playing the same kind of "in-name" readability role in config. Rather, they (again, to me) suggest hierarchy rather than readability enhancements.